optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-client/src/main/java/org/optaplanner/workbench/screens/domaineditor/client/widgets/planner/DataObjectFieldPickerItemLabelViewImpl.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.workbench.screens.domaineditor.client.widgets.planner;
+
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.ClickHandler;
+import com.google.gwt.event.shared.HandlerRegistration;
+import com.google.gwt.user.client.ui.Composite;
+import org.jboss.errai.common.client.dom.Span;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.EventHandler;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+
+@Dependent
+@Templated
+public class DataObjectFieldPickerItemLabelViewImpl extends Composite implements DataObjectFieldPickerItemLabelView {
+
+    @Inject
+    @DataField("name")
+    Span name;
+
+    @Inject
+    @DataField("removeIcon")
+    Span removeIcon;
+
+    @Override
+    public HandlerRegistration addClickHandler( ClickHandler clickHandler ) {
+        return addHandler( clickHandler, ClickEvent.getType() );
+    }
+
+    @Override
+    public void setName( String name ) {
```

Comment:
IMO either choose to have ```final``` parameters or not.. but using a mix-match is inconsistent.